core/templates/pages/admin-page/features-tab/admin-features-tab.component.ts

```diff
@@ -0,0 +1,249 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the feature tab in the admin panel.
+ */
+
+import { Component, OnInit, Input } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+import { AdminFeaturesTabConstants } from
+  './admin-features-tab.constants';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { AdminDataService } from
+  'pages/admin-page/services/admin-data.service';
+import { AdminTaskManagerService } from
+  'pages/admin-page/services/admin-task-manager.service';
+import { PlatformParameter, FeatureStage } from
+  'domain/platform_feature/platform-parameter-object.factory';
+import { PlatformFeatureAdminBackendApiService } from
+  'domain/platform_feature/platform-feature-admin-backend-api.service';
+
+import cloneDeep from 'lodash/cloneDeep';
+
+
+import { PlatformParameterRuleObjectFactory, PlatformParameterRule } from
+  'domain/platform_feature/platform-parameter-rule-object.factory';
+import {
+  PlatformParameterFilterType,
+  PlatformParameterFilterObjectFactory,
+  PlatformParameterFilter,
+  ServerMode,
+} from 'domain/platform_feature/platform-parameter-filter-object.factory';
+
+@Component({
+  selector: 'admin-features-tab',
+  templateUrl: './admin-features-tab.directive.html'
+})
+export class AdminFeaturesTabComponent implements OnInit {
+  @Input() setStatusMessage: (msg: string) => void;
+
+  readonly availableFilterTypes: PlatformParameterFilterType[] = Object
+    .keys(PlatformParameterFilterType)
+    .map(key => PlatformParameterFilterType[key]);
+
+  readonly filterTypeToContext: {
```

Comment:
Nice implementation design. This is a good use of generic programming to make things simpler & more declarative.